pyk/src/pyk/proof/reachability.py

```diff
@@ -86,7 +86,6 @@ class APRProof(Proof[APRProofStep, APRProofResult], KCFGExploration):
     error_info: Exception | None
     prior_loops_cache: dict[int, list[int]]
 
-    _checked_for_terminal: set[int]
```

Comment:
Given how proof execution is designed, I don't think there's a possibility of repeated terminalness checks for the same node, so I think we could lose this piece of state. Even if repeated checks do happen, they are very fast.